src/main/java/de/muenchen/isi/domain/service/AbfrageService.java

```diff
@@ -82,7 +82,17 @@ public InfrastrukturabfrageModel saveInfrastrukturabfrage(final Infrastrukturabf
      */
     public InfrastrukturabfrageModel updateInfrastrukturabfrage(final InfrastrukturabfrageModel abfrage) throws EntityNotFoundException, UniqueViolationException {
         this.getInfrastrukturabfrageById(abfrage.getId());
-        return this.saveInfrastrukturabfrage(abfrage);
+        var abfrage4update = this.abfrageDomainMapper.model2entity(abfrage);
+        var saved = this.infrastrukturabfrageRepository.findByAbfrage_NameAbfrageIgnoreCase(abfrage4update.getAbfrage().getNameAbfrage());
+
+        if (saved.isEmpty()) {
+            return this.saveInfrastrukturabfrage(abfrage);
+        } else if (abfrage4update.getId().equals(saved.get().getId())) {
+            abfrage4update = this.infrastrukturabfrageRepository.save(abfrage4update);
```

Comment:
Evtl kann die Prüflogik generischer (Auf Basis der Optional<T extends BaseEntity>, da nur die Id geprüft wird.) gestaltet und wie ursprünglich implementiert in der save-Methode verwendet werden.

Die Prüflogik prüft ob ein Speichern möglich ist und gibt entsprechend true oder false zurück. Ist ein Speichern möglich, so wird gespeichert. Ist kein speichern möglich, so wird die in der Save-Methode die Exception geworfen.